src/FaucetV1.sol

```diff
@@ -0,0 +1,103 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.0;
+
+interface IERC20 {
+    function transferFrom( address src, address dst, uint256 rawAmount) external returns (bool);
+    function balanceOf(address account) external view returns (uint256);
+}
+
+/// A RAD Faucet
+contract FaucetV1 {
+    address public owner;
+
+    uint256 public maxWithdrawAmount;
+    uint256 public maxConsecutiveWithdrawals;
+
+    struct Withdrawal {
+        uint256 timestamp;
+        uint256 amount;
+    }
+
+    mapping(address => Withdrawal) public lastWithdrawalByUser;
+    mapping(address => uint256) public consecutiveWithdrawals;
+
+    constructor(
+        uint256 _maxConsecutiveWithdrawals,
+        uint256 _maxWithdrawAmount
+    ) {
+        owner = msg.sender;
+        maxConsecutiveWithdrawals = _maxConsecutiveWithdrawals;
+        maxWithdrawAmount = _maxWithdrawAmount;
+    }
+
+    event OwnershipTransferred(address indexed previousOwner, address indexed newOwner);
+    event Transfer( address indexed _from, address indexed _to, uint256 indexed _amount);
+
+    modifier onlyOwner() {
+        require(msg.sender == owner, "Faucet: Only the faucet owner can perform this action.");
+        _;
+    }
+
+    /// @notice Sets a new owner to the faucet
+    /// @param _newOwner Address of new owner
+    function setOwner(address _newOwner) public onlyOwner {
+        require(_newOwner != address(0), "Faucet: New owner is the zero address");
+        address _oldOwner = owner;
+        owner = _newOwner;
+
+        emit OwnershipTransferred(_oldOwner, _newOwner);
+    }
+
+    /// @notice Calculates the the amount of hours to wait, relative to the amount withdrawed.
+    /// @param _amount The amount of tokens to be withdrawed
+    /// @return The number of NFTs owned by `_owner`, possibly zero
+    function calculateTimeLock(uint256 _amount) public pure returns (uint256) {
+        return _amount**2 * 1 / 10;
+    }
+
+    /// @notice Sets the max amount available to withdraw in one withdrawal.
+    /// @param _amount The max amount of tokens to be withdrawed
+    function setMaxAmount(uint256 _amount) public onlyOwner {
+        maxWithdrawAmount = _amount;
+    }
+
+    // @notice Sets the maximum amount of consecutive withdrawals, before one has to wait 24 hours.
+    /// @param _amount Amount of consecutive withdrawals allowed
+    function setMaxConsecutiveWithdrawals(uint256 _amount) public onlyOwner {
+        maxConsecutiveWithdrawals = _amount;
+    }
+
+    /// @notice Calculates the the amount of hours to wait, relative to the amount withdrawed.
+    /// @param _requester The address of the user requesting tokens
+    /// @return Bool if requester is allowed to create more withdrawals
+    function checkConsecutiveWithdrawals(address _requester) private view returns (bool) {
+        return consecutiveWithdrawals[_requester] < maxConsecutiveWithdrawals;
+    }
```

Comment:
Can we avoid checking for consecutive withdrawals completely, and just check for the amount per time unit? I'm not sure we gain anything from limiting the number of times someone withdraws, since they pay for the gas anyway.

Reply:
We have two ways of restricting withdrawals.
- One is per time via a quadratic function, that timelocks the withdrawal function for a specific time.
- The other is to limit consecutive withdrawals in 24 hours.

The reason behind it is that without, one could make small amounts every few seconds and like that slowly drain the faucet..

Reply:
Hmm, I get it, but I think it would be simpler and cheaper if we prevented that with a single time-based cap, no? Eg. `min(timeSinceLastWithdrawal, 1 day) * multiplier`

Reply:
Hmm I think we could do a single time based cap, and eventually raise the minimum amount to wait, to for example 30 minutes?

I'm not quite sure what you meant with the multiplier in your example.

Reply:
Hmm I think we could do a single time based cap, to save on deployment costs.

Since the user will be paying for the gas, we could just remove the consecutive withdrawals and leave the time based cap as it is currently.
Eventually the user will be spending more Rinkeby ETH on gas fees, for obtaining less to nothing on Rinkeby RAD.

Reply:
Yeah, that's what I'm thinking!